### DISCUSSION TITLE
Async call C++ functions

### DISCUSSION THREAD

**Discussion by zliucd** [pybind/pybind11#3447] [Q&A]
Hi,

The C++ function may take some time(from seconds to minutes) to execute. Is there a way to **call C++ function asynchronously** using `Pybind`?  e.g., timer, callback. 

Thanks.

**Comment by jiwaszki** [comment]
Try this: https://pybind11.readthedocs.io/en/stable/advanced/misc.html#global-interpreter-lock-gil
